.github/workflows/ci.build.yml

```diff
@@ -0,0 +1,214 @@
+            -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
+            install \
+            -U \
+            -DskipTests=true \
+            -Dmaven.javadoc.skip=false \
+            -Dgpg.skip \
+            -f $ROOT_POM
+      - name: Generate hashes
+        shell: bash
+        id: hash
+        if: matrix.mode == 'JRE'
+        run: |
+          echo "Building SLSA provenance material..."
+          ls guava/target/*.jar guava-gwt/target/*.jar guava-testlib/target/*.jar
+          echo "hashes=$(sha256sum guava/target/*.jar guava-gwt/target/*.jar guava-testlib/target/*.jar | base64 -w0)" >> ./provenance-hashes.txt
+          cat ./provenance-hashes.txt >> "$GITHUB_OUTPUT"
+          echo "Gathered provenance hashes:"
+          cat ./provenance-hashes.txt
+      - name: 'Upload artifacts'
+        uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4.3.1
+        if: matrix.mode == 'JRE'
+        with:
+          name: guava-artifacts-${{ matrix.mode == 'Android' && 'android' || 'jre' }}-${{ github.sha }}
+          path: |
+            guava/target/*.jar
+            guava-gwt/target/*.jar
+            guava-testlib/target/*.jar
+            ./provenance-hashes.txt
+          if-no-files-found: warn
+          retention-days: 7
+
+  # Generate SLSA provenance
+  provenance:
+    needs: [build]
+    if: inputs.provenance
+    name: "SLSA Provenance"
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.9.0
+    permissions:
+      actions: read
+      id-token: write
+      contents: write
+    with:
+      base64-subjects: "${{ needs.build.outputs.hashes }}"
+      upload-assets: ${{ inputs.provenance_publish }}
+
+  # Publish snapshot JAR
+  publish_snapshot:
+    name: 'Publish Snapshot'
+    needs: [build, provenance]
+    if: inputs.snapshot
+    runs-on: ubuntu-latest
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@63c24ba6bd7ba022e95695ff85de572c04a18142 # v2.7.0
+        with:
+          egress-policy: audit
+      - name: 'Check out repository'
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - name: 'Set up JDK 21'
+        uses: actions/setup-java@9704b39bf258b59bc04b50fa2dd55e9ed76b47a8 # v4.1.0
+        with:
+          java-version: 21
+          distribution: 'zulu'
+          server-id: ${{ inputs.repository }}
+          server-username: CI_DEPLOY_USERNAME
+          server-password: CI_DEPLOY_PASSWORD
+          cache: 'maven'
+      - name: "Download artifacts"
+        uses: actions/download-artifact@c850b930e6ba138125429b7e5c93fc707a7f8427 # v4.1.4
+        with:
+          name: guava-artifacts-jre-${{ github.sha }}
+      - name: 'Publish'
+        env:
+          CI_DEPLOY_USERNAME: ${{ secrets.CI_DEPLOY_USERNAME }}
+          CI_DEPLOY_PASSWORD: ${{ secrets.CI_DEPLOY_PASSWORD }}
+        run: ./util/deploy_snapshot.sh
```

Comment:
@sgammon Thanks for opening this PR. It would be great to harden the build pipelines for Guava artifacts. I think one issue with the current workflow though is that only the snapshot artifacts are automatically published and not the main artifacts. So, the SLSA provenances can only attest to the snapshot artifacts. We also need to automate the publishing of main artifacts to Maven Central, and generate SLSA provenances for them.

Reply:
@behnazh-w Yes, you're right. I wanted to get things started and churning, and then, with good PR feedback, I am happy to extend the SLSA provenance changes to other parts of the workflow. I believe there are Maven and Bazel generators that could be put to good use.